packages/arb-bridge-eth/contracts/GlobalInbox.sol

```diff
@@ -99,7 +118,9 @@ contract GlobalInbox is GlobalEthWallet, GlobalFTWallet, GlobalNFTWallet, IGloba
             if (!valid) {
                 return (false, startOffset);
             }
-            transferEth(msg.sender, to, value);
+
+            address paymentOwner = getPaymentOwner(to, nodeHash, messageIndex);
+            transferEth(msg.sender, paymentOwner, value);
```

Comment:
Can we give back the storage used for the payment record here, after the transfer succeeds?  This might save a little bit of gas.  But only do it if it's definitely safe--better to waste a little gas than risk dropping a payment.